src/redux/features/observations/observationsThunks.ts

```diff
@@ -1,61 +1,26 @@
 import { Dispatch } from 'redux';
 import {
-  ObservationResults,
-  ObservationPlantingZoneResults,
-  ObservationPlantingSubzoneResults,
+  ObservationResultsPayload,
+  ObservationPlantingZoneResultsPayload,
+  ObservationPlantingSubzoneResultsPayload,
   ObservationMonitoringPlotResults,
 } from 'src/types/Observations';
 import { ObservationsService } from 'src/services';
 import { RootState } from 'src/redux/rootReducer';
 import { setObservationsResultsAction } from './observationsSlice';
-import getDateDisplayValue from '@terraware/web-components/utils/date';
-
-export type PlantingSiteTimeZone = {
-  id: number;
-  timeZone?: string;
-};
-
-/**
- * Apply time zones to completed time dates
- * TODO: apply them to other dates as we see-fit (for now completed times are the only used dates)
- */
-const applyTimeZone = (observation: ObservationResults, timeZone?: string): ObservationResults => {
-  return {
-    ...observation,
-    completedTime: observation.completedTime ? getDateDisplayValue(observation.completedTime, timeZone) : undefined,
-    plantingZones: observation.plantingZones.map((zone: ObservationPlantingZoneResults) => ({
-      ...zone,
-      completedTime: zone.completedTime ? getDateDisplayValue(zone.completedTime, timeZone) : undefined,
-      plantingSubzones: zone.plantingSubzones.map((subZone: ObservationPlantingSubzoneResults) => ({
-        ...subZone,
-        monitoringPlots: subZone.monitoringPlots.map((monitoringPlot: ObservationMonitoringPlotResults) => ({
-          ...monitoringPlot,
-          completedTime: monitoringPlot.completedTime
-            ? getDateDisplayValue(monitoringPlot.completedTime, timeZone)
-            : undefined,
-        })),
-      })),
-    })),
-  };
-};
 
 /**
- * Fetch observation results and post-process completed times to match planting site time zones
+ * Fetch observation results
  */
-export const requestObservationsResults = (organizationId: number, plantingSiteTimeZones: PlantingSiteTimeZone[]) => {
+export const requestObservationsResults = (organizationId: number) => {
   return async (dispatch: Dispatch, _getState: () => RootState) => {
     try {
       const response = await ObservationsService.listObservationsResults(organizationId);
       const { error, observations } = response;
       dispatch(
         setObservationsResultsAction({
           error,
-          observations: observations.map((observation) =>
-            applyTimeZone(
-              observation,
-              plantingSiteTimeZones.find((site) => observation.plantingSiteId === site.id)?.timeZone
-            )
-          ),
+          observations,
```

Comment:
Removed the `applyTimeZone` call because we're getting the timezone from the planting site?

Reply:
Correct, this code was moved into the selector. Previously the client of this request had to pass the planting site timezones into the request. Now, the selector has access to planting sites and their timezones.